### PR TITLE
Re-add .NET Framework to tests and other cleanup

### LINKED ISSUE #1490: Re-enable desktop tests

### PR DESCRIPTION
**Pull request by natemcmaster**
Re-add .NET Framework testing. Resolves #1490 
Remove workarounds
Unify dependencies versions

### CODE REVIEW

File: test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests.csproj [hunk 1]
```diff
@@ -3,10 +3,9 @@
   <Import Project="..\..\build\common.props" />
 
   <PropertyGroup>
-    <TargetFrameworks>netcoreapp1.1</TargetFrameworks>
+    <TargetFrameworks>netcoreapp1.1;net46</TargetFrameworks>
```
Comment (cesarbs) [inline test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests.csproj]: `net462`?

File: test/Microsoft.AspNetCore.Server.KestrelTests/Microsoft.AspNetCore.Server.KestrelTests.csproj [hunk 2]
```diff
@@ -3,11 +3,10 @@
   <Import Project="..\..\build\common.props" />
 
   <PropertyGroup>
-    <TargetFrameworks>netcoreapp1.1</TargetFrameworks>
+    <TargetFrameworks>netcoreapp1.1;net46</TargetFrameworks>
```
Comment (cesarbs) [inline test/Microsoft.AspNetCore.Server.KestrelTests/Microsoft.AspNetCore.Server.KestrelTests.csproj]: `net462`?

**Comment by davidfowl** [comment]
Did you fix the hang?

**Comment by natemcmaster** [comment]
It didn't reproduce the 3rd time i ran tests. And appears to be fine on CI.

File: test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests.csproj (hunk 1 above)
Comment (natemcmaster) [inline test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests.csproj]: We only have net46 on our ci agents.

**Comment by cesarbs** [comment]
What hung?

**Comment by natemcmaster** [comment]
The net46 tests appeared to hang the first time they ran. On further inspection, its not that they hang. They are just 20 times slower. KestrelTests runs in 30 seconds on .net core and 7 minutes on .net framework.

**Comment by cesarbs** [comment]
😱 

That doesn't sound right. Could it be an issue in the test tooling? Otherwise we have to find the offending tests.

**Comment by natemcmaster** [comment]
It's the way we use the test framework. The test framework has to de/serialize thousands of tests across the app domain boundary. Disabling appdomains seems to fix the perf problem, but causes tests to fail with:
```
 System.TypeInitializationException : The type initializer for 'Microsoft.AspNetCore.Server.KestrelTests.HttpsConnectionAdapterTests' threw an exception.
---- System.Security.Cryptography.CryptographicException : The system cannot find the file specified.


Stack Trace:
   at Microsoft.AspNetCore.Server.KestrelTests.HttpsConnectionAdapterTests..ctor()
----- Inner Stack Trace -----
   at System.Security.Cryptography.CryptographicException.ThrowCryptographicException(Int32 hr)
   at System.Security.Cryptography.X509Certificates.X509Utils._QueryCertFileType(String fileName)
   at System.Security.Cryptography.X509Certificates.X509Certificate.LoadCertificateFromFile(String fileName, Object password, X509KeyStorageFlags keyStorageFlags)
   at System.Security.Cryptography.X509Certificates.X509Certificate2..ctor(String fileName, String password)
   at Microsoft.AspNetCore.Server.KestrelTests.HttpsConnectionAdapterTests..cctor() in C:\dev\KestrelHttpServer\test\Microsoft.AspNetCore.Server.KestrelTests\HttpsConnectionAdapterTests.cs:line 27
```

**Comment by davidfowl** [comment]
3rd time's a charm 😄

**Comment by natemcmaster** [comment]
:up: :date: disabled appdomains in KestrelTests. Also, re-enabled section of code that was conditionally compiled desktop .net.

**Comment by cesarbs** [comment]
Why `net46` and not `net462`?

**Comment by natemcmaster** [comment]
We only have net46 on our ci agents.

**Comment by natemcmaster** [comment]
Disabling app domains on desktop brings the test run from 7 minutes to 48 seconds. Still 15 seconds slower than the .NET Core run :-/

**Pull request merged**